addons/paint-snap/ui.js

```diff
@@ -0,0 +1,287 @@
+import {
+  eventTarget,
+  enable,
+  disable,
+  addonIsEnabled,
+  setSnapFrom,
+  setSnapTo,
+  snapOn,
+  snapFrom,
+  snapTo,
+  toggle,
+  threshold,
+  setThreshold,
+} from "./state.js";
+
+/** @type {import("../../addon-api/content-script/typedef").UserscriptUtilities} */
+export function initUI({ addon, msg }) {
+  const createGroup = () => {
+    const el = document.createElement("div");
+    el.className = "sa-paint-snap-group";
+    return el;
+  };
+  const createSeperator = () => {
+    const el = document.createElement("div");
+    el.className = "sa-paint-snap-settings-seperator";
+    return el;
+  };
+
+  const createButton = ({ useButtonTag } = {}) => {
+    const el = document.createElement(useButtonTag ? "button" : "span");
+    el.className = "sa-paint-snap-button";
+    el.setAttribute("role", "button");
+    return el;
+  };
+
+  const createButtonImage = (name) => {
+    const el = document.createElement("img");
+    el.className = "sa-paint-snap-image";
+    el.draggable = false;
+    el.src = addon.self.dir + "/icons/" + name + ".svg";
+    return el;
+  };
+
+  const controlsGroup = createGroup();
+  addon.tab.displayNoneWhileDisabled(controlsGroup, {
+    display: "flex",
+  });
+
+  const settingPageWrapper = document.createElement("div");
+  settingPageWrapper.className = "sa-paint-snap-settings-wrapper";
+  controlsGroup.appendChild(settingPageWrapper);
+
+  const settingsPage = document.createElement("div");
+  settingsPage.className = "sa-paint-snap-settings";
+  settingsPage.dataset.compat = addonIsEnabled("onion-skinning");
```

Comment:
Instead of this, make `onion-skinning` detect this addon. Do not use `getEnabledAddons` for this. (I'll probably make this crash if this is called from other addons at some point.)

Reply:
I dont even need this, will remove. Sorry. Also will remove `window.paper = paper`